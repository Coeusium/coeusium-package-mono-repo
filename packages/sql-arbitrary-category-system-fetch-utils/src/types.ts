export type { CategorySet, Category } from 'sql-arbitrary-category-system';
