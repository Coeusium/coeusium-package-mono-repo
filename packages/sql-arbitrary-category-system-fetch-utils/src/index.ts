export * from './sql-arbitrary-category-system-fetch-utils';
export type { Category, CategorySet } from './types';
