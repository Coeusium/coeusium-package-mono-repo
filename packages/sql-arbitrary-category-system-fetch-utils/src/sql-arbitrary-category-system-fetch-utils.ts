import { Category, CategorySet } from './types';
import { almFetch } from '@almaclaine/fetch-utils';

// Category Set

export async function addCategorySet(name) {
  const res = await almFetch(`/api/category/set?name=${name}`, {
    method: 'post',
  });
  const { id } = await res.json();
  return id as string;
}

export async function getCategorySet(id) {
  const res = await almFetch(`/api/category/set?id=${id}`);
  return (await res.json()) as CategorySet;
}

export async function listCategorySets(page = 0, limit = 20) {
  const res = await almFetch(`/api/category/set?page=${page}&limit=${limit}`);
  return (await res.json()) as CategorySet[];
}

export async function deleteCategorySet(id) {
  const res = await almFetch(`/api/category/set?id=${id}`, {
    method: 'delete',
  });
  await res.json();
}

// Category

export async function addCategory(name, set_id, parent = '') {
  let url = `/api/category?name=${name}&set_id=${set_id}`;
  if (parent) url = `${url}&parent=${parent}`;
  const res = await almFetch(url, { method: 'post' });
  const { id } = await res.json();
  return id as string;
}

export async function getCategory(id) {
  const res = await almFetch(`/api/category?id=${id}`);
  return (await res.json()) as Category;
}

export async function getCategoryChildren(id) {
  const res = await almFetch(`/api/category?id=${id}&type=children`);
  return (await res.json()) as Category[];
}

export async function getTopLevelCategories(id) {
  const res = await almFetch(`/api/category?id=${id}&type=top`);
  return (await res.json()) as Category[];
}

export async function getCategoryDescendants(id) {
  const res = await almFetch(`/api/category?id=${id}&type=descendants`);
  return (await res.json()) as Category[];
}

export async function deleteCategory(id) {
  const res = await almFetch(`/api/category?id=${id}`, { method: 'delete' });
  await res.json();
}
