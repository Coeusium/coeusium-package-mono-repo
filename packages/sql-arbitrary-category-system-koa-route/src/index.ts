export * from './sql-arbitrary-category-system-koa-route';
export type { ConnectionInfo, Category, CategorySet } from './types';
