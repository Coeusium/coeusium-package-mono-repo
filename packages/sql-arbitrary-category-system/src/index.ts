export * from './sql-arbitrary-category-system';
export type { ConnectionInfo, Category, CategorySet } from './types';
