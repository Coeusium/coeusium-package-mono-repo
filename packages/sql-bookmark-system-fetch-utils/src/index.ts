export * from './sql-bookmark-system-fetch-utils';
export type { Bookmark } from './types';
