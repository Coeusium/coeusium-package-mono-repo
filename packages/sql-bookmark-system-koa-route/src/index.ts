export * from './sql-bookmark-system-koa-route';
export type { ConnectionInfo, Bookmark } from './types';
