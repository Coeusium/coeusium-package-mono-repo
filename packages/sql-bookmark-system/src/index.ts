export * from './sql-bookmark-system';
export type { ConnectionInfo, Bookmark } from './types';
