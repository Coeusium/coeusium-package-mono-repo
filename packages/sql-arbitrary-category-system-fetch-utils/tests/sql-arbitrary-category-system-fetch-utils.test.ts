import * as sqlArbitaryTagSystemFetchUtils from '../src';

describe('sql-arbitrary-category-system-fetch-utils', () => {
  it('Has Correct API', () => {
    const keys = Object.keys(sqlArbitaryTagSystemFetchUtils);
    expect(keys.length).toBe(10);
    expect(keys).toMatchSnapshot();
  });
});
