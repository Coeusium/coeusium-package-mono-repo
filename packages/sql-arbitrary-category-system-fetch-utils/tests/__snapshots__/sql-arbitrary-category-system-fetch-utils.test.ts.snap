// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sql-arbitrary-category-system-fetch-utils Has Correct API 1`] = `
Array [
  "addCategorySet",
  "getCategorySet",
  "listCategorySets",
  "deleteCategorySet",
  "addCategory",
  "getCategory",
  "getCategoryChildren",
  "getTopLevelCategories",
  "getCategoryDescendants",
  "deleteCategory",
]
`;
